ebs-recover-and-replace/src/ebs_recover_and_replace/__init__.py

```diff
@@ -934,6 +934,248 @@ def load_plan(load_file):
+            InstanceIds=[plan_instance_id]
+        )
+
+    except botocore.exceptions.UnauthorizedSSOTokenError as ssotokenerr:
+        sys.exit(ssotokenerr)
+
+    except botocore.exceptions.ProfileNotFound as profileerr:
+        sys.exit(profileerr)
+
+    if response["ResponseMetadata"]["HTTPStatusCode"] == 200:
+        for reservation in response["Reservations"]:
+            if len(reservation["Instances"]) == 1:
+                print("FOUND")
+                for instance in reservation["Instances"]:
+                    print("    Verifying Instance Name... ",
+                        end="",
+                        flush=True,
+                    )
+                    instance_name = ""
+                    for tags in instance["Tags"]:
+                        if tags["Key"] == "Name":
+                            instance_name = tags["Value"]
+
+                    if plan_instance_metadata["Name"] == instance_name:
+                        print("OK")
+                    else:
+                        print("FAILED")
+                        sys.exit("Error: Instance name does not match\nPlan: {}, Actual: {}".format(
+                            plan_instance_metadata["Name"],
+                            instance_name
+                        ))
+
+                    print("    Verifying Instance IP Address... ",
+                        end="",
+                        flush=True,
+                    )
+                    if plan_instance_metadata["IPAddress"] == instance["PrivateIpAddress"]:
+                        print("OK")
+                    else:
+                        print("FAILED")
+                        sys.exit("Error: Instance IP Address does not match\nPlan: {}, Actual: {}".format(
+                            plan_instance_metadata["IPAddress"],
+                            instance["PrivateIpAddress"]
+                        ))
+
+            else:
+                print("FAILED")
+                sys.exit("Error: The instance with Instance ID {} could not be found".format(plan_instance_id))
+
+    else:
+        print("FAILED")
+        print(
+            "Received {} error from AWS".format(
+                response["ResponseMetadata"]["HTTPStatusCode"]
+            )
+        )
+
+
+def verify_volume(ec2_client, plan_volume_id, plan_volume_metadata):
+    """
+    Verifies that the provided Volume exists in AWS
+    and that the provided volume data matches
+    """
+
+    print("Checking Volume ID: {}... ".format(plan_volume_id),
+        end="",
+        flush=True,
+    )
+
+    try:
+        response = ec2_client.describe_volumes(
+            VolumeIds=[plan_volume_id]
+        )
+
+    except botocore.exceptions.UnauthorizedSSOTokenError as ssotokenerr:
+        sys.exit(ssotokenerr)
+
+    except botocore.exceptions.ProfileNotFound as profileerr:
+        sys.exit(profileerr)
+
+    if response["ResponseMetadata"]["HTTPStatusCode"] == 200:
```

Comment:
I'd be tempted to create a common function called something like `validate_response` which exits on invalid responses. We can simplify some of these follow on checks then to be focussed purely on their specific logic. i.e. if any volumes were found etc

Reply:
Updated in [8d3d35d](https://github.com/companieshouse/platform-tools-python/pull/3/commits/8d3d35d536771e2f178ccef8700da7c65b9d7794)